pydicom/filebase.py

```diff
@@ -30,19 +30,13 @@ def __init__(self, *args, **kwargs):
     def read_le_tag(self):
         """Read and return two unsigned shorts (little endian)
            from the file."""
-        bytes_read = self.read(4)
-        if len(bytes_read) < 4:
-            # needed for reading "next" tag when at end of file
-            raise EOFError
+        bytes_read = self.read(4, need_exact_length=True)
         return unpack(b"<HH", bytes_read)
 
     def read_be_tag(self):
```

Comment:
We could change the "API" with deprecation... I put "API" in quotes as I don't think this would have been used directly in user code, so I feel it is not an important API.  So we could change to a more descriptive name (and update the name everywhere in pydicom code), and alias the old name to it with deprecation warning.  I hate to say it, but as in so many issues/PRs, if it is ever going to happen, then v1.0 is the time.